apps/services/user-profile/src/app/user-profile/userProfile.controller.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Put,
+  NotFoundException,
+  Param,
+  Post,
+  NotImplementedException,
+} from '@nestjs/common'
+import {
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiParam,
+  ApiTags,
+} from '@nestjs/swagger'
+import { CreateUserProfileDto } from './dto/createUserProfileDto'
+import { UpdateImageDto } from './dto/updateImageDto'
+import { UpdateUserProfileDto } from './dto/updateUserProfileDto'
+import { UserProfileByNationalIdPipe } from './pipes/userProfileByNationalId.pipe'
+import { UserProfile } from './userProfile.model'
+import { UserProfileService } from './userProfile.service'
+
+@ApiTags('User Profile')
+@Controller()
+export class UserProfileController {
+  constructor(
+    private userProfileService: UserProfileService
+  ) { }
+
+  @Get('userProfile/:nationalId')
+  @ApiOkResponse({ type: UserProfile })
+  async findOneByNationalId(
+    @Param('nationalId') nationalId: string,
+  ): Promise<UserProfile> {
+    const profile = await this.userProfileService.findByNationalId(nationalId)
+
+    if (!profile) {
+      throw new NotFoundException(
+        `A User profile with the nationalId ${nationalId} does not exist`,
+      )
+    }
+
+    return profile
+  }
+
+  @Post('userProfile/')
+  @ApiCreatedResponse({ type: UserProfile })
+  async create(
+    @Body()
+    application: CreateUserProfileDto,
+  ): Promise<UserProfile> {
+    return await this.userProfileService.create(application)
+  }
+
+  @Put('userProfile/:nationalId')
```

Comment:
I encourage you to add tests for GET, PUT and POST.

Reply:
Was on the agenda. But if not now then when :)